### ISSUE TITLE
Bluetooth Controllers & Hakchi 2.15

### ISSUE THREAD

**Issue by magneek** [ClusterM/hakchi2#307]
hello folks,

I'm having trouble with the functionality of both NES30 Pro & SNES30 controllers while playing SNES and Sega Genesis games on my modded NES Classic. for some reason the A & B functions repeat on the X & Y buttons and the L & R buttons. any ideas?

**Comment by magneek** [comment]
I have both sets of controllers and the NES Classic Bluetooth receiver updated to the latest firmware. I'm able to play NES games just fine, with the 8bitdo controllers. If plug in a Wii Classic controller, that works fine as well with Sega Genesis and SNES games. However if I load up a SNES/Genesis game and try to use the 8bitdo controllers I don't have the additional functionality of the X,Y,L & R buttons.

**Comment by NaikiTurincondeliphone** [comment]
Config the buttons in Retroarch "Select + Start" and Setting/input

The firmware for 8Bitdo Receiver NES Mini must be 1.05 Beta 0

**Comment by magneek** [comment]
maybe that's the problem, I definitely updated the receiver to the most recent/latest update. I will update the receiver to the firmware mentioned above and try again.

**Comment by evilbert70** [comment]
re assigning buttons in retro arch is not working it allows you to reassign y and b and a but you reassign x and tell me what happens!!!!! it displays search and locks up that's what happens... super Mario world is not playable with the 8bitdo receiver plus ps4 controller.... I know I have tried... I have also tried several firmware versions but only 1.05 beta 0 and 103 will actually allow the receiver to boot up... I have given up on the wireless until hakchi has gotten further along. there are many people who want blutooth ps4 support ... but for now we wait....

**Comment by evilbert70** [comment]
Naki try to set up an 8bit do for yourself and then try as you suggest to reassign the buttons in retro arch... it does not work period!!!!!

**Comment by magneek** [comment]
thanks folks, I was able to get my NES30 Pro and NES Retro Receiver to work properly on my NES Classic while running SNES, Sega Genesis and Neo Geo using the firmware described above. NES30 Pro is using FW 1.72 and my RR is using FW 1.05 beta 0.

thanks for the help, although I would like to see a version of Hakchi that will allow me to use the most recent updates on my controllers and receivers.

**Comment by alexzea26** [comment]
HI magneek, wich version of hakchi do you use? i have the same problem

**Comment by magneek** [comment]
AlexZea: i'm using Hakchi 2.15, once i updated my controller and receiver the proper FW (mentioned above) i was able to play the other modules just fine. does this help?

**Comment by alexzea26** [comment]
magneek: thank you very much for your help. The only issue its, i need to instal the firmware "RR is using FW 1.05 beta 0." for the retro reciver mini? and my NES30 PRO controler will be ok?

http://forum.8bitdo.com/forum.php?mod=attachment&aid=NTMxfDhmYjU2NWM5fDE0OTMwNzE4NDJ8MHwxNDA0

**Comment by magneek** [comment]
you'll need the correct update for the NES30 Pro as well, if you have FW other than 1.72 you'll run into the same issue I had. once  you have FW 1.72 on the NES30 Pro and FW 1.05 beta on the receiver you'll be able to play NES games but also Sega, SNES, Turbo Grafx, etc. games without running into a button mapping issue.

**Comment by alexzea26** [comment]
magneek: thank you again for your time and help. I got it, but another question:

Which version of firmware i need for the control? 

1.72 with date 02-28 
http://download.8bitdo.com/Firmware/Pro/Firmware_FC30&NES30_Pro_V1.72.zip

or 

1.72 **## beta0** with date 02-23
http://forum.8bitdo.com/forum.php?mod=attachment&aid=NDk0fDhhMDBlNzlkfDE0OTMxMzY3MDZ8MHwxMTc5

**Comment by magneek** [comment]
good question, try them both out. I'm not too sure which one I was using...I thought there was only option for the 1.72 FW. it's not a problem to update to one FW then change the FW to another. or at least I didn't have an issue with that. I had the most recent FW then went down to the 1.72 FW without a problem.

**Comment by alexzea26** [comment]
magneek: thank you very very very much, with 1.72 beta, 1.05 beta0, hakchi 2.16 working, my son and me are very greatful with you, my two controlers NES30 PRO are awesome

**Comment by magneek** [comment]
that's awesome! glad I could help! now you can have fun from the sofa playing all the classics! enjoy!

**Comment by jimmythronx** [comment]
Good evening. I purchased the retro recieviers made by 8bitdo but just the receivers because I have the Wii u pro controller and ps4 joysticks. With the nes games I can play perfectly fine but when I try to configure it with the snes games I cannot do it. I updated the last FW from 8bitdo web site 01/05/2017 unfortunately after this it still doesn't work . Any help? because I think the nes30 controller works fine as far as I can read above. But Wii u and ps4 controllers? thanks for your answers and greetings from Peru.

**Comment by magneek** [comment]
jimmy thronx, sorry pal, i'm not sure how you'll correct this. I suspect an update is needed on the hakchi end since you're not going to be able to change the FW on a PS4 or Wii U controllers...but then again i'm not sure.

**Comment by thrashology** [comment]
I am also having a hard time mapping out a ps4 controller for use with the 8bit receiver. I wonder if any one has figured it out yet? I was able to get the sega geneses to map out 2 different buttons but cant get the 3rd button.... when i tried to map out controllers for n64 using my ps4 controller it also was not working.

**Comment by vatw** [comment]
I wanted to say that I too had a similar issue-- bluetooth adapter seemed to work fine with the stock kernel, but after updating, I would drop connection every 2-3 minutes (very annoying).
Firmware update of the receiver helped, but still happened.

Then I replaced power supply I was using, and all troubles went away-- just adding this here for future reference -- If firmware update doesn't work, and you still have bluetooth (or similar) dongle issues, try installing a better power supply (vs. using an existing USB port on random device near where you installed, as I did initially)
